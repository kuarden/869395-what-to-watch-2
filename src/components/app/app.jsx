import React from "react";
import PropTypes from 'prop-types';
import Main from '../main/main.jsx';
import {connect} from 'react-redux';
import {ActionCreator} from "../../reducer/reducer";

export class App extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {films, genres, onChangeGenre, onGetFilms} = this.props;
    return <Main films={films} genres={genres} onChangeGenre={onChangeGenre} onGetFilms={onGetFilms} />;
  }
}

App.propTypes = {
  films: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        genre: PropTypes.string,
        image: PropTypes.string,
        url: PropTypes.string,
      })
  ).isRequired,
  genres: PropTypes.arrayOf(PropTypes.string),
  onChangeGenre: PropTypes.func.isRequired,
  onGetFilms: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const res = Object.assign({}, ownProps, {
    genre: state.activeItem,
    genres: state.genres,
    films: state.films,
  });

  return res;
};

const mapDispatchToProps = (dispatch) => ({
  onChangeGenre: (genre) => {
    dispatch(ActionCreator.changeGenre(genre));
    dispatch(ActionCreator.getFilms());
  },
  onGetFilms: () => dispatch(ActionCreator.getFilms())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);

