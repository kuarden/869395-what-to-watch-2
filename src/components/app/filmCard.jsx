import React, {Fragment} from 'react';
import {PropTypes} from 'prop-types';

const FilmCard = (props) => {

  const {data} = props;

  return <Fragment>
    <article className="small-movie-card catalog__movies-card">
      <div className="small-movie-card__image">
        <img src={data.image} alt={data.name} width="280" height="175"/>
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">{data.name}</a>
      </h3>
    </article>
  </Fragment>;
};

FilmCard.propTypes = {
  data: PropTypes.object.isRequired
};

export default FilmCard;
