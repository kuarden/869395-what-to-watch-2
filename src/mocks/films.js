export const films = [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    image: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    name: `Macbeth`,
    image: `img/macbeth.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 4,
    name: `Aviator`,
    image: `img/aviator.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    name: `We need to talk about Kevin`,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 6,
    name: `What We Do in the Shadows`,
    image: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 7,
    name: `Revenant`,
    image: `img/revenant.jpg`,
    genre: `Drama`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 8,
    name: `Johnny English`,
    image: `img/johnny-english.jpg`,
    genre: `Comedy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 9,
    name: `Shutter Island`,
    image: `img/shutter-island.jpg`,
    genre: `Thriller`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 10,
    name: `Pulp Fiction`,
    image: `img/pulp-fiction.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 11,
    name: `No Country for Old Men`,
    image: `img/no-country-for-old-men.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 12,
    name: `Snatch`,
    image: `img/snatch.jpg`,
    genre: `Action`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 13,
    name: `Moonrise Kingdom`,
    image: `img/moonrise-kingdom.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 14,
    name: `Seven Years in Tibet`,
    image: `img/seven-years-in-tibet.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 15,
    name: `Midnigth Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 16,
    name: `War of the Worlds`,
    image: `img/war-of-the-worlds.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 17,
    name: `Dardjeeling Limited`,
    image: `img/dardjeeling-limited.jpg`,
    genre: `Adventures`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 18,
    name: `Orlando`,
    image: `img/orlando.jpg`,
    genre: `Adventures`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 19,
    name: `Mindhunter`,
    image: `img/mindhunter.jpg`,
    genre: `Action`,
    url: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 20,
    name: `Midnight Special`,
    image: `img/midnight-special.jpg`,
    genre: `Fantasy`,
    url: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];
