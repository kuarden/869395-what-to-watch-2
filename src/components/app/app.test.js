import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import {App} from "./app.jsx";
import {films} from "../../mocks/films.js";

Enzyme.configure({adapter: new Adapter()});

it(`App renders correctly`, () => {
  const tree = shallow(<App
    films={films}
    onChangeGenre = {jest.fn()}
    onGetFilms = {jest.fn()}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
