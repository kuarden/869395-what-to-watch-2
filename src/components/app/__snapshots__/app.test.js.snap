// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Main
  films={
    Array [
      Object {
        "genre": "Fantasy",
        "id": 1,
        "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "name": "Fantastic Beasts: The Crimes of Grindelwald",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Drama",
        "id": 2,
        "image": "img/bohemian-rhapsody.jpg",
        "name": "Bohemian Rhapsody",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Fantasy",
        "id": 3,
        "image": "img/macbeth.jpg",
        "name": "Macbeth",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Adventures",
        "id": 4,
        "image": "img/aviator.jpg",
        "name": "Aviator",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Adventures",
        "id": 5,
        "image": "img/we-need-to-talk-about-kevin.jpg",
        "name": "We need to talk about Kevin",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Fantasy",
        "id": 6,
        "image": "img/what-we-do-in-the-shadows.jpg",
        "name": "What We Do in the Shadows",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Drama",
        "id": 7,
        "image": "img/revenant.jpg",
        "name": "Revenant",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Comedy",
        "id": 8,
        "image": "img/johnny-english.jpg",
        "name": "Johnny English",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Thriller",
        "id": 9,
        "image": "img/shutter-island.jpg",
        "name": "Shutter Island",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Action",
        "id": 10,
        "image": "img/pulp-fiction.jpg",
        "name": "Pulp Fiction",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Action",
        "id": 11,
        "image": "img/no-country-for-old-men.jpg",
        "name": "No Country for Old Men",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Action",
        "id": 12,
        "image": "img/snatch.jpg",
        "name": "Snatch",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Fantasy",
        "id": 13,
        "image": "img/moonrise-kingdom.jpg",
        "name": "Moonrise Kingdom",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Adventures",
        "id": 14,
        "image": "img/seven-years-in-tibet.jpg",
        "name": "Seven Years in Tibet",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Fantasy",
        "id": 15,
        "image": "img/midnight-special.jpg",
        "name": "Midnigth Special",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Fantasy",
        "id": 16,
        "image": "img/war-of-the-worlds.jpg",
        "name": "War of the Worlds",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Adventures",
        "id": 17,
        "image": "img/dardjeeling-limited.jpg",
        "name": "Dardjeeling Limited",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Adventures",
        "id": 18,
        "image": "img/orlando.jpg",
        "name": "Orlando",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
      Object {
        "genre": "Action",
        "id": 19,
        "image": "img/mindhunter.jpg",
        "name": "Mindhunter",
        "url": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      },
      Object {
        "genre": "Fantasy",
        "id": 20,
        "image": "img/midnight-special.jpg",
        "name": "Midnight Special",
        "url": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      },
    ]
  }
  onChangeGenre={[MockFunction]}
  onGetFilms={[MockFunction]}
/>
`;
